### ISSUE TITLE
Normalizing PNGs: Zlib::BufError: buffer error

### ISSUE THREAD

**Issue by maxsz** [jruby/jruby#598]
# Intro

Apple uses some pre-compilation (basically byte swapping) on PNG files stored inside an iOS app. To reverse these changes, there exists code like this: https://github.com/swcai/iphone-png-normalizer/blob/master/iphonepng.rb. Running this particular script with MRI succeeds, but fails with JRuby for _some_ images.
# Error

Particularly with this image (![Icon-114](https://f.cloud.github.com/assets/540025/284915/46c96086-9206-11e2-98d6-7b2bd33b1692.png)) I get the following error. Note that this image appears to be broken here, because of the byte swapping mentioned above. Also note that MRI (1.9.3) doesn't choke on it.

```
Zlib::BufError: buffer error
         finish at org/jruby/ext/zlib/ZStream.java:110
     decompress at normalizePNG.rb:28
      normalize at normalizePNG.rb:79
            map at org/jruby/RubyArray.java:2361
      normalize at normalizePNG.rb:76
  normalize_png at normalizePNG.rb:111
           open at org/jruby/RubyIO.java:1183
  normalize_png at normalizePNG.rb:110
         (root) at normalizePNG.rb:120
           each at org/jruby/RubyArray.java:1613
         (root) at normalizePNG.rb:120
           each at org/jruby/RubyArray.java:1613
         (root) at normalizePNG.rb:120
```

**Comment by MSNexploder** [comment]
This still fails with JRuby 1.7.18
```
jruby 1.7.18 (1.9.3p551) 2014-12-22 625381c on Java HotSpot(TM) 64-Bit Server VM 1.8.0_25-b17 [darwin-x86_64]
test/68747470733a2f2f662e636c6f75642e6769746875622e636f6d2f6173736574732f3534303032352f3238343931352f34366339363038362d393230362d313165322d393864362d3762326264333362313639322e706e67.png
Zlib::BufError: buffer error
         finish at org/jruby/ext/zlib/ZStream.java:134
     decompress at iphonepng.rb:28
      normalize at iphonepng.rb:79
        collect at org/jruby/RubyArray.java:2399
            map at org/jruby/RubyArray.java:2412
      normalize at iphonepng.rb:76
  normalize_png at iphonepng.rb:111
           open at org/jruby/RubyIO.java:1181
  normalize_png at iphonepng.rb:110
         (root) at iphonepng.rb:120
           each at org/jruby/RubyArray.java:1613
         (root) at iphonepng.rb:120
           each at org/jruby/RubyArray.java:1613
         (root) at iphonepng.rb:120
```

but works on master

```
jruby 9.0.0.0-SNAPSHOT (2.2.0p0) 2015-01-01 fb9199f Java HotSpot(TM) 64-Bit Server VM 25.25-b02 on 1.8.0_25-b17 +jit [darwin-x86_64]
test/68747470733a2f2f662e636c6f75642e6769746875622e636f6d2f6173736574732f3534303032352f3238343931352f34366339363038362d393230362d313165322d393864362d3762326264333362313639322e706e67.png
Corrupted PNG file
Done.
```

**Comment by enebo** [comment]
This works in 9k and we will never fix this in 1.7.x as it is winding down....resolving.
